src/Mod/Path/PathScripts/PathVcarve.py

```diff
@@ -0,0 +1,280 @@
+        obj.setEditorMode('BaseShapes', 2)  # hide
+        if not hasattr(obj, 'BaseObject'):
+            obj.addProperty("App::PropertyLink", "BaseObject", "Path",
+                            QtCore.QT_TRANSLATE_NOOP("PathVcarve",
+                            "Additional base objects to be engraved"))
+        obj.setEditorMode('BaseObject', 2)  # hide
+
+    def initOperation(self, obj):
+        '''initOperation(obj) ... create vcarve specific properties.'''
+        obj.addProperty("App::PropertyFloat", "Discretize", "Path",
+                        QtCore.QT_TRANSLATE_NOOP("PathVcarve",
+                        "The deflection value for discretizing arcs"))
+        obj.addProperty("App::PropertyFloat", "Threshold", "Path",
+                        QtCore.QT_TRANSLATE_NOOP("PathVcarve",
+                        "cutoff for removing colinear segments (degrees). \
+                        default=10.0."))
+        obj.addProperty("App::PropertyFloat", "Tolerance", "Path",
+                QtCore.QT_TRANSLATE_NOOP("PathVcarve", ""))
+        obj.Threshold = 10.0
+        obj.Discretize = 0.01
+        obj.Tolerance = PathPreferences.defaultGeometryTolerance()
+        self.setupAdditionalProperties(obj)
+
+    def opOnDocumentRestored(self, obj):
+        # upgrade ...
+        self.setupAdditionalProperties(obj)
+
+    def buildPathMedial(self, obj, Faces):
+        '''constructs a medial axis path using openvoronoi'''
+
+        def insert_many_wires(vd, wires):
+            for wire in wires:
+                PathLog.debug('discretize value: {}'.format(obj.Discretize))
+                pts = wire.discretize(QuasiDeflection=obj.Discretize)
+                ptv = [FreeCAD.Vector(p[0], p[1]) for p in pts]
+                ptv.append(ptv[0])
+
+                for i in range(len(pts)):
+                    vd.addSegment(ptv[i], ptv[i+1])
+
+        def calculate_depth(MIC):
+            # given a maximum inscribed circle (MIC) and tool angle,
+            # return depth of cut.
+            maxdepth = obj.ToolController.Tool.CuttingEdgeHeight
+            toolangle = obj.ToolController.Tool.CuttingEdgeAngle
+            d = round(MIC / math.tan(math.radians(toolangle / 2)), 4)
+            return d if d <= maxdepth else maxdepth
+
+        def getEdges(vd, color=[0]):
+            if type(color) == int:
+                color = [color]
+            geomList = []
+            for e in vd.Edges:
+                if e.Color not in color:
+                    continue
+                if e.toGeom(8) is None:
+                    continue
+                p1 = e.Vertices[0].toGeom(calculate_depth(0-e.getDistances()[0]))
+                p2 = e.Vertices[-1].toGeom(calculate_depth(0-e.getDistances()[-1]))
+                newedge = Part.Edge(Part.Vertex(p1), Part.Vertex(p2))
+
+                newedge.fixTolerance(obj.Tolerance, Part.Vertex)
+                geomList.append(newedge)
+
+            if geomList:
+                return geomList
```

Comment:
need to return `geomList` even when it's empty - `None` isn't handled by the logic further down.

Reply:
fixed